fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/erase/EraseTestMain.java

```diff
@@ -0,0 +1,211 @@
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.database.utils.api.IConnectionProvider;
+import com.ibm.fhir.database.utils.api.IDatabaseAdapter;
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.database.utils.common.JdbcPropertyAdapter;
+import com.ibm.fhir.database.utils.common.JdbcTarget;
+import com.ibm.fhir.database.utils.db2.Db2Adapter;
+import com.ibm.fhir.database.utils.db2.Db2PropertyAdapter;
+import com.ibm.fhir.database.utils.derby.DerbyAdapter;
+import com.ibm.fhir.database.utils.derby.DerbyPropertyAdapter;
+import com.ibm.fhir.database.utils.model.DbType;
+import com.ibm.fhir.database.utils.postgres.PostgresAdapter;
+import com.ibm.fhir.database.utils.postgres.PostgresPropertyAdapter;
+import com.ibm.fhir.database.utils.postgres.PostgresTranslator;
+import com.ibm.fhir.persistence.ResourceEraseRecord;
+import com.ibm.fhir.persistence.erase.EraseDTO;
+import com.ibm.fhir.persistence.jdbc.connection.FHIRDbFlavor;
+import com.ibm.fhir.persistence.jdbc.connection.FHIRDbFlavorImpl;
+import com.ibm.fhir.persistence.jdbc.dao.EraseResourceDAO;
+
+/**
+ * EraseTestMain is a test driver for the EraseResourceDAO so it can be debugged during development.
+ * input is ResourceType LogicalId
+ * Main Patient 1791f19a126-28b0239c-f1a1-4d15-979f-920af8d12446
+ *
+ * Example output:
+ * <code>
+ * May 12, 2021 1:36:39 PM com.ibm.fhir.persistence.jdbc.test.erase.EraseTestMain createConnection
+ * INFO: Opening connection to: jdbc:postgresql://localhost:5432/fhirdb?currentSchema=fhirdata&schemaName=fhirdata&ssl=false
+ * Got a Connection
+ * Starting the erase of 'Patient/1791f19a126-28b0239c-f1a1-4d15-979f-920af8d12446'
+ * ResourceEraseRecord [total=-1, status=NOT_FOUND]
+ * </code>
+ */
+public class EraseTestMain {
+    private static final String CLASSNAME = EraseTestMain.class.getSimpleName();
+    private static final Logger logger = Logger.getLogger(CLASSNAME);
+
+    private IDatabaseTranslator translator = new PostgresTranslator();
+    private DbType dbType = DbType.POSTGRESQL;
+    private Properties inProperties = null;
+    private Properties dbProperties = null;
+    private String[] args = null;
+
+    private String schemaName = "fhirdata";
+    private String resourceType = null;
+    private String logicalId = null;
+
+
+    public EraseTestMain(Properties properties, String[] args) {
+        this.args = args;
+        this.inProperties = properties;
+    }
+
+    protected void erase() throws Exception {
+        try {
+            try (Connection c = createConnection()) {
+                System.out.println("Got a Connection");
+                try {
+                    FHIRDbFlavor flavor = new FHIRDbFlavorImpl(dbType, true);
+                    EraseResourceDAO dao = new EraseResourceDAO(c, translator, schemaName, flavor, null, null);
```

Comment:
I guess if you switch to using FHIRPersistenceJDBCCache to obtain the resource_type_id, you'd also need to instantiate and initialize it in this test before invoking the DAO.

Reply:
I've added some mock to inject the cache.

Reply:
Changed in next commit